data/Store.js

```diff
@@ -138,25 +150,26 @@ export class Store {
         }
 
         // 2) Pre-process summary record, peeling it out of updates if needed
-        let {summaryRecord} = this,
-            summaryUpdateRec;
+        const {summaryRecord} = this;
+        let summaryUpdateRec;
         if (summaryRecord) {
             [summaryUpdateRec] = lodashRemove(updateRecs, {id: summaryRecord.id});
             if (!summaryUpdateRec && rawSummaryData) {
-                summaryUpdateRec = this.createRecord(rawSummaryData);
+                summaryUpdateRec = this.createRecord({...this.summaryRecord.raw, ...rawSummaryData}, null, true);
             }
         }
 
         // 3) Apply changes
         let didUpdate = false;
         if (!isEmpty(updateRecs) || (addRecs && addRecs.size) || !isEmpty(remove)) {
-            this._all = this._all.updateData({update: updateRecs, add: addRecs, remove: remove});
+            this._original = this._original.loadRecordTransaction({update: updateRecs, add: addRecs, remove: remove});
```

Comment:
if original and all were the same when this method was called, aren't we doing double the work, and saving an extra representation?

Reply:
We could consider not allowing the loading of a transaction for a dirty recordset.